plugin/gluoncv_backend.py

```diff
@@ -0,0 +1,439 @@
+        make_dir(self.get_local_path(self.validation_uri))
+        self.validation_zip_uri = join(base_uri, 'validation.zip')
+
+        self.scratch_uri = join(base_uri, 'scratch')
+        make_dir(self.get_local_path(self.scratch_uri))
+
+    def download(self):
+        def _download(data_zip_uri):
+            data_zip_path = self.download_if_needed(data_zip_uri)
+            data_dir = os.path.splitext(data_zip_path)[0]
+            shutil.unpack_archive(data_zip_path, data_dir)
+
+        _download(self.training_zip_uri)
+        _download(self.validation_zip_uri)
+
+    def upload(self):
+        def _upload(data_uri):
+            data_dir = self.get_local_path(data_uri)
+            shutil.make_archive(data_dir, 'zip', data_dir)
+            self.upload_or_copy(data_uri + '.zip')
+
+        _upload(self.training_uri)
+        _upload(self.validation_uri)
+
+
+class ModelFiles(FileGroup):
+    """Utilities for files produced when calling train."""
+
+    def __init__(self, base_uri, tmp_dir, replace_model=False):
+        """Create these model files.
+
+        Args:
+            base_uri: Base URI of the model files
+            replace_model: If the model file exists, remove.
+                           Used for the training step, to retrain
+                           existing models.
+
+        Returns:
+            A new ModelFile instance.
+        """
+        FileGroup.__init__(self, base_uri, tmp_dir)
+        self.model_uri = join(self.base_uri, 'model')
+        self.log_uri = join(self.base_uri, 'log.csv')
+
+        if replace_model:
+            if os.path.exists(self.model_uri):
+                os.remove(self.model_uri)
+            if os.path.exists(self.log_uri):
+                os.remove(self.log_uri)
+
+    def get_backend_config(self, pretrained_model_uri, gcv_config,
+                           dataset_files, class_map):
+
+        from collections import defaultdict
+        config = defaultdict(lambda: defaultdict(dict))
+
+        # Update config using local paths.
+        config['trainer']['options']['output_dir'] = \
+            self.get_local_path(self.base_uri)
+        config['model']['model_path'] = self.get_local_path(self.model_uri)
+        config['model']['nb_classes'] = len(class_map)
+        config['trainer']['options']['training_data_dir'] = \
+            dataset_files.get_local_path(dataset_files.training_uri)
+        config['trainer']['options']['validation_data_dir'] = \
+            dataset_files.get_local_path(dataset_files.validation_uri)
+        config['trainer']['options']['class_names'] = \
+            class_map.get_class_names()
+
+        # Save the pretrained weights locally
+        pretrained_model_path = None
+        if pretrained_model_uri:
+            pretrained_model_path = self.download_if_needed(
+                pretrained_model_uri)
+
+        return (config, pretrained_model_path)
+
+
+class GluonCVBackend(Backend):
```

Comment:
I would move the actual backend and associated functions into a separate file to mimic the way the other backends are implemented (ie. separate `<backend>_config.py` and `<backend>.py` files)